### PR TITLE
Support ENS name on non-mainnet

### PR DESCRIPTION
**Pull request by nkrulikovsky**
https://github.com/saddle-finance/saddle-frontend/issues/795

### CODE REVIEW

File: src/hooks/useENS.ts [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+        chainId === ChainId.MAINNET
+      ) {
         const provider = new ethers.providers.Web3Provider(library.provider)
         const ensName = await provider.lookupAddress(address)
+        console.log("ens name on net", ensName)
         if (ensName) setENSName(ensName)
+      } else if (
+        address &&
+        library &&
+        ethers.utils.isAddress(address) &&
```
Comment (ug02fast) [inline src/hooks/useENS.ts:27]: not necessary but putting this logic into a const could clean up a couple lines.

File: src/hooks/useENS.ts [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
+        chainId !== ChainId.MAINNET
+      ) {
+        // Get ens name from main chain when network is non-mainnet
+        const ensNameOnMainnet = await ethers
+          .getDefaultProvider()
```
Comment (ug02fast) [inline src/hooks/useENS.ts:32]: What is the difference between .getDefaultProvider() and Web3Provider.

**Review by ug02fast — APPROVED** [review]
looks good just left a couple comments

File: src/hooks/useENS.ts [hunk 3]
```diff
@@ -1,23 +1,40 @@
...
```
Comment (hammeiam) [inline src/hooks/useENS.ts:31]: Can we use REACT_APP_NETWORK_URL as an argument here? My concern is that `getDefaultProvider` basically spams free endpoints, meaning that it creates a ton of network requests and will eventually return 429s or something like that. Here's was the network looks like when moving to another chain: 

Screen Shot 2022-01-27 at 7 15 54 PM

Comment (nkrulikovsky) [reply src/hooks/useENS.ts:31]: You are right.

**Review by ug02fast — APPROVED** [review]

**Review by hammeiam — APPROVED** [review]
Works well, thank you Nikita.eth 🤠

**Pull request merged**